ui/src/app/pages/login/account-creation/account-creation.spec.tsx

```diff
@@ -36,16 +44,59 @@ function findCountryInputField() {
   return screen.getByLabelText('Country input') as HTMLInputElement;
 }
 
+function getAreaOfResearchTextBox(container: HTMLElement) {
+  return container.querySelector('textarea#areaOfResearch');
+}
+
 const defaultConfig = { gsuiteDomain: 'researchallofus.org' };
 
 beforeEach(() => {
   serverConfigStore.set({ config: defaultConfig });
   registerApiClient(ProfileApi, new ProfileApiStub());
-  props = {
-    profile: createEmptyProfile(),
-    onComplete: () => {},
-    onPreviousClick: () => {},
-  };
+});
+
+it('should allow completing the account creation form', async () => {
+  const user = userEvent.setup();
+  const onComplete = jest.fn();
+  const { container } = component({
+    ...createProps(),
+    onComplete,
+  });
+
+  await user.type(screen.getByLabelText(FORM_LABELS.username), 'username');
+  await user.type(screen.getByLabelText(FORM_LABELS.givenName), 'Firstname');
+  await user.type(screen.getByLabelText(FORM_LABELS.familyName), 'Lastname');
+
+  await user.type(
+    screen.getByLabelText(FORM_LABELS.streetAddress1),
+    '1 Main Street'
+  );
+  await user.type(screen.getByLabelText(FORM_LABELS.city), 'Boston');
+  await user.type(findStateField(), 'MA');
+  await user.type(screen.getByLabelText(FORM_LABELS.zipCode), '02115');
+  await user.click(findCountryDropdownField());
+  await user.paste(countryDropdownOption.unitedStates);
+  await user.keyboard('{enter}');
+
+  await user.click(getAreaOfResearchTextBox(container));
+  await user.paste('I am an undergraduate learning genomics.');
+
+  await user.click(screen.getByLabelText('Next'));
+
+  expect(onComplete).toHaveBeenCalled();
+  const profile = onComplete.mock.calls[0][0];
```

Comment:
what is this magic?

Reply:
`onComplete` = jest stub
`.mock` = ✨ 
`.calls[0]` = get the first call to the stub
`[0]` = get the first argument

There's a jest helper (`expect...toBeCalledWith` or similar) that requires matching the entire object or defining jest "matchers" for the other arguments. Since the Profile object has a ton of fields, I personally prefer this approach.